omod/src/test/java/org/openmrs/module/fhir2/narrative/DefaultNarrativeTemplateTest.java

```diff
@@ -0,0 +1,230 @@
+		
+		AllergyIntolerance.AllergyIntoleranceReactionComponent allergyIntoleranceReactionComponent1 = new AllergyIntolerance.AllergyIntoleranceReactionComponent();
+		allergyIntoleranceReactionComponent1.addManifestation().addCoding().setCode("64305001").setDisplay("Urticaria");
+		allergyIntoleranceReactionComponent1.setOnset(new Date());
+		allergyIntoleranceReactionComponent1.setSeverity(AllergyIntolerance.AllergyIntoleranceSeverity.fromCode("moderate"));
+		allergyIntoleranceReactionComponent1.addNote()
+		        .setText("The patient reports that the onset of urticaria was within 15 minutes of eating cashews.");
+		allergyIntolerance.addReaction(allergyIntoleranceReactionComponent1);
+		
+		assertThat(testHelper(new AllergyIntolerance()).getDivAsString(), notNullValue());
+		assertThat(testHelper(allergyIntolerance).getDivAsString(), notNullValue());
+	}
+	
+	@Test
+	public void testForMedicationRequest() {
+		// https://www.hl7.org/fhir/medicationrequest0301.html
+		
+		MedicationRequest medicationRequest = new MedicationRequest();
+		
+		medicationRequest.setId("medrx301");
+		
+		CodeableConcept medicationCode = new CodeableConcept();
+		medicationCode.addCoding().setSystem("http://snomed.info/sct").setDisplay("Oral Form Oxycodone (product)")
+		        .setCode("430127000");
+		medicationRequest.addContained(new Medication().setCode(medicationCode).setId("med0310"));
+		Provenance provenance = new Provenance();
+		provenance.addTarget().setReference("ServiceRequest/physiotherapy");
+		provenance.setRecorded(new Date());
+		Provenance.ProvenanceAgentComponent provenanceAgentComponent = new Provenance.ProvenanceAgentComponent();
+		provenanceAgentComponent.addRole().addCoding().setCode("AUT")
+		        .setSystem("http://terminology.hl7.org/CodeSystem/v3-ParticipationType");
+		provenanceAgentComponent.setWho(new Reference().setDisplay("Dr Adam Careful").setReference("Practitioner/example"));
+		provenance.addAgent(provenanceAgentComponent);
+		Signature signature = new Signature();
+		signature.addType().setSystem("urn:iso-astm:E1762-95:2013").setCode("1.2.840.10065.1.12.1.1")
+		        .setDisplay("Author's Signature");
+		signature.setWhen(new Date());
+		signature.setWho(new Reference().setReference("Practitioner/example").setDisplay("Dr Adam Careful"));
+		signature.setTargetFormat("application/fhir+xml");
+		signature.setSigFormat("application/signature+xml");
+		signature.setDataElement(new Base64BinaryType("dGhpcyBibG9iIGlzIHNuaXBwZWQ="));
+		provenance.addSignature(signature);
+		medicationRequest.addContained(provenance);
+		
+		medicationRequest.addIdentifier().setUse(Identifier.IdentifierUse.OFFICIAL)
+		        .setSystem("http://www.bmc.nl/portal/prescriptions").setValue("12345689");
+		medicationRequest.setStatus(MedicationRequest.MedicationRequestStatus.COMPLETED);
+		
+		CodeableConcept statusReason = new CodeableConcept();
+		statusReason.addCoding().setDisplay("Try another treatment first").setCode("altchoice")
```

Comment:
@ibacher what do you think of @ShivanshRakesh   arranging his data into data sets ?

Reply:
@HerbertYiga If by this you mean "should this data be stored in a file and loaded from there" my answer would be yes. This should be a different mechanism from loading data into the database, if only because there are a number of layers between the raw data in the database and the narrative that we don't need to worry about here.